acq4/devices/FilterWheel/filterwheel.py

```diff
@@ -175,12 +175,8 @@ def _getPosition(self):
 
     def loadPreset(self, name):
         """Load a preset filter wheel position by name."""
-        idx = None
-        for i, n in self._slotNames.items():
-            if n == name:
-                idx = i
-                break
-        self.setPosition(idx)
+        idx = next((i for i, n in self._slotNames.items() if n == name), None)
+        return self.setPosition(idx)
```

Comment:
The `return` was the only important part here. My editor shortened the for loop.